code/modules/vote/vote_datum.dm

```diff
@@ -0,0 +1,192 @@
+	switch(vote_result_type)
+		if(VOTE_RESULT_TYPE_MAJORITY)
+			var/list/results = list()
+
+			// Count up all votes
+			for(var/ck in voted)
+				if(voted[ck] in results)
+					results[voted[ck]]++
+				else
+					results[voted[ck]] = 1
+
+			// Get the biggest vote count, since we can also use this to pick tiebreaks
+			var/maxvotes = 0
+			for(var/res in results)
+				if(results[res] > maxvotes)
+					maxvotes = results[res]
+
+			var/list/winning_options = list()
+
+			for(var/res in results)
+				if(results[res] == maxvotes)
+					winning_options |= res
+
+			if(!length(winning_options))
+				to_chat(world, "<span class='interface'>No votes were cast. Do you all hate democracy?!</span>") // shame them
+				return null
+
+			// Print all results
+			for(var/res in results)
+				if(res in winning_options)
+					// Make it stand out
+					to_chat(world, "<span class='info'><code>[res]</code> - [results[res]] vote\s</span>")
+				else
+					// Make it normal
+					to_chat(world, "<span class='interface'><code>[res]</code> - [results[res]] vote\s</span>")
+
+				// And log it to the DB
+				if(!is_custom)
+					SSblackbox.record_feedback("nested tally", "votes", results[res], list(vote_type_text, res), ignore_seal = TRUE)
+
+			if(length(winning_options) > 1)
+				var/random_dictator = pick(winning_options)
+				to_chat(world, "<span class='interface'><b>Its a tie between [english_list(winning_options)]. Picking <code>[random_dictator]</code> at random.</b></span>") // shame them
+				return random_dictator
+
+			// If we got here there must only be one thing in the list
+			var/res = winning_options[1]
+			to_chat(world, "<span class='interface'><b><code>[res]</code> won the vote.</b></span>") // shame them
+			return res
+
+
+/datum/vote/proc/announce(start_text)
+	to_chat(world, {"<font color='purple'><b>[start_text]</b>
+		<a href='?src=[SSvote.UID()];vote=open'>Click here or type <code>Vote</code> to place your vote.</a>
+		You have [GLOB.configuration.vote.vote_time / 10] seconds to vote.</font>"})
+	SEND_SOUND(world, sound('sound/ambience/alarm4.ogg'))
+
+
+/datum/vote/proc/tick()
+	if(remaining() == 0)
+		// Announce result
+		var/result = calculate_result()
+		handle_result(result)
+		qdel(src)
+
+
+/datum/vote/Destroy(force)
+	// Should always be true but ehhhhhhh
+	if(SSvote.active_vote == src)
+		SSvote.active_vote = null
+	return ..()
+
+
+// Override on children
+/datum/vote/proc/handle_result(result)
+	return
+
+/datum/vote/proc/generate_choices()
+	return
+
+
+/*
+	UI STUFFS
+*/
+/datum/vote/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = FALSE, datum/tgui/master_ui = null, datum/ui_state/state = GLOB.always_state)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "VotePanel", "VotePanel", 400, 500, master_ui, state)
+		ui.open()
+
+/datum/vote/ui_data(mob/user)
+	var/list/data = list()
+	data["remaining"] = remaining()
```

Comment:
I'm not sure what this looks like in-game but this may need to be rounded to a whole second

Reply:
In some cases it can show as `54.9`, but I dont see that as a major issue.

If I do round it ill do it JS side to take load off the server.